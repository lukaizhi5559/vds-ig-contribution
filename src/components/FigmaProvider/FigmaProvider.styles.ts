/*!
 * Copyright (C) Verizon. All rights reserved.
 */

import { createUseStyles } from 'react-jss';

/**
 * Styles for the FigmaProvider slots
 */
export const useFigmaProviderStyles = createUseStyles({
  root: {},
});
