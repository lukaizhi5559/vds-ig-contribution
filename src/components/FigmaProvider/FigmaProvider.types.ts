/*!
 * Copyright (C) Verizon. All rights reserved.
 */

/**
 *  FigmaProviderProps
 */
export type FigmaProviderProps = {
  children?: React.ReactNode;
};
