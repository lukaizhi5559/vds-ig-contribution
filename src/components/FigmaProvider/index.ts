/*!
 * Copyright (C) Verizon. All rights reserved.
 */

export * from './FigmaProvider';
export * from './FigmaProvider.types';
export * from './FigmaProvider.styles';
