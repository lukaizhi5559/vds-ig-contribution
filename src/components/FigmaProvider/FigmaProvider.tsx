/*!
 * Copyright (C) Verizon. All rights reserved.
 */

import * as React from 'react';

import type { FigmaProviderProps } from './FigmaProvider.types';
import { useFigmaProviderStyles } from './FigmaProvider.styles';

/**
 * Render the final JSX of FigmaProvider
 */
export const FigmaProvider: React.FC<FigmaProviderProps> = (props: FigmaProviderProps) => {
  const { children } = props;
  const styles = useFigmaProviderStyles();

  return <div className={styles.root}>{children}</div>;
};
