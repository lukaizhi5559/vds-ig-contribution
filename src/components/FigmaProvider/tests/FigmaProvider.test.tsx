/*!
 * Copyright (C) Verizon. All rights reserved.
 */

import * as React from 'react';
import { render } from '@testing-library/react';
import { axe, toHaveNoViolations } from 'jest-axe';

import { FigmaProvider } from './../FigmaProvider';

expect.extend(toHaveNoViolations);

describe('FigmaProvider', () => {
  it('should not have a11y violations', async () => {
    expect.assertions(1);
    const { container } = render(<FigmaProvider>Default FigmaProvider A11y</FigmaProvider>);
    expect(await axe(container)).toHaveNoViolations();
  });

  it('renders a default state', () => {
    expect.assertions(1);
    const { container } = render(<FigmaProvider>Default FigmaProvider</FigmaProvider>);
    expect(container).toMatchSnapshot();
  });
});
