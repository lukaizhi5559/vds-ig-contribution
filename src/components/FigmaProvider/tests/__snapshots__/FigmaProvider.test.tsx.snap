// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FigmaProvider renders a default state 1`] = `
<div>
  Default FigmaProvider
</div>
`;
